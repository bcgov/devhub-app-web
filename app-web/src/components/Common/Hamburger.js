import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import { ARIA_LABEL_FILTER_SELECT } from '../../constants/strings';
const Hamburger = ({ clicked, ...rest }) => (
  <FontAwesomeIcon
    {...rest}
    icon={faBars}
    onClick={clicked}
    size="2x"
    aria-label={ARIA_LABEL_FILTER_SELECT}
  />
);

Hamburger.propTypes = {
  clicked: PropTypes.func.isRequired,
};

export default Hamburger;
