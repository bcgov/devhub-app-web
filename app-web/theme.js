/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING } from './src/constants/designTokens';
import { RESOURCE_TYPES, TOPICS, SEARCH_RESOURCE_TYPES, JOURNEY } from './src/constants/ui';

export const DEVHUB_PALETTE = {
  [RESOURCE_TYPES.COMPONENTS]: '#4299D2',
  [RESOURCE_TYPES.DOCUMENTATION]: '#246BD1',
  [RESOURCE_TYPES.PEOPLE]: '#E17039',
  [RESOURCE_TYPES.SELF_SERVICE_TOOLS]: '#F3BA45',
  [TOPICS]: '#444',
  [RESOURCE_TYPES.REPOSITORIES]: '#6e5494',
  [SEARCH_RESOURCE_TYPES.GITHUB_ISSUE]: '#6e5494',
  [RESOURCE_TYPES.EVENTS]: '#f44b42',
  [JOURNEY]: '#009688',
  lightgreen: '#66bb6a',
  yellow: '#FCBA19',
  lightyellow: '#F3BA45',
  grey: '#494949',
  lightgrey: '#F2F2F2',
  darkgrey: '#444',
  orange: '#E17039',
  purple: '#6E5494',
  red: '#D8292F',
  lightred: '#F44B42',
  green: '#2E8540',
  blue: '#003366',
  babyblue: '#4299D2',
  lightblue: '#246BD1',
  linkblue: '#1A5A96',
  bgBlue: '#38598A',
  white: '#ffffff',
};

export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
