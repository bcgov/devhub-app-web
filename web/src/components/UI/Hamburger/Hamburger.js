import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import { ARIA_LABEL_FILTER_SELECT } from '../../../constants/ariaLabels';

export const TEST_IDS = {
  hamburger: 'hamburger-icon-button',
};

const Hamburger = ({ clicked, ...rest }) => (
  <FontAwesomeIcon
    {...rest}
    icon={faBars}
    onClick={clicked}
    data-testid={TEST_IDS.hamburger}
    size="2x"
    aria-label={ARIA_LABEL_FILTER_SELECT}
  />
);

Hamburger.propTypes = {
  clicked: PropTypes.func.isRequired,
};

export default Hamburger;
