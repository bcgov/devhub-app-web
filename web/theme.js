/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING, DEVHUB_PALETTE } from './src/constants/designTokens';
export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
