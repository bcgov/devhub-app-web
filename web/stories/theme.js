import { create } from '@storybook/theming';
import { DEVHUB_PALETTE } from '../theme';

export default create({
  base: 'light',

  colorPrimary: 'hotpink',
  colorSecondary: 'deepskyblue',

  // UI
  appBg: 'white',
  appContentBg: 'white',
  appBorderColor: 'grey',
  appBorderRadius: 4,

  // Typography
  fontBase: 'sans-serif',
  fontCode: 'monospace',

  // Text colors
  textColor: DEVHUB_PALETTE.darkgrey,
  textInverseColor: 'rgba(255,255,255,0.9)',

  // Toolbar default and active colors
  barTextColor: DEVHUB_PALETTE.white,
  barSelectedColor: 'black',
  barBg: DEVHUB_PALETTE.blue,

  // Form colors
  inputBg: 'white',
  inputBorder: 'silver',
  inputTextColor: 'black',
  inputBorderRadius: 4,

  brandTitle: 'My custom storybook',
  brandImage:
    'https://u58tqm22fk-flywheel.netdna-ssl.com/wp-content/uploads/2016/06/BC-Gov-Logo-CMYK_pos.png',
});
