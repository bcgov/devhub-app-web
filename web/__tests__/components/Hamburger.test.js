/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import theme from '../../theme';
import Hamburger, { TEST_IDS } from '../../src/components/UI/Hamburger/Hamburger';

describe('Hamburger Component', () => {
  const props = {
    clicked: jest.fn(),
    className: 'foo',
  };

  test('it matches snapshot', () => {
    const { container, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Hamburger {...props} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
    const hamburger = getByTestId(TEST_IDS.hamburger);
    fireEvent.click(hamburger);
    expect(props.clicked).toHaveBeenCalled();
  });
});
