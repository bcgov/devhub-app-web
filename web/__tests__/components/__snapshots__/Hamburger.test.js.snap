// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Component it matches snapshot 1`] = `
<svg
  aria-hidden="true"
  aria-label="Select to filter cards by the resource type"
  class="svg-inline--fa fa-bars fa-w-14 fa-2x foo"
  data-icon="bars"
  data-prefix="fas"
  data-testid="hamburger-icon-button"
  focusable="false"
  role="img"
  viewBox="0 0 448 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
    fill="currentColor"
  />
</svg>
`;
